vpc/service/prune_unused_ips.go

```diff
@@ -0,0 +1,64 @@
+package service
+
+import (
+	"context"
+	"database/sql"
+
+	"github.com/Netflix/titus-executor/logger"
+	"github.com/Netflix/titus-executor/vpc/tracehelpers"
+	"github.com/pkg/errors"
+	"go.opencensus.io/trace"
+)
+
+const (
+	//  Each Titus task requires an IPv4 address. Each VPC has 3 x /16s for Titus to operate in.
+	// That means that the maximum number of "interesting" IPv4 rows per table are 3*(2**16) = 196608
+	howManyUnusedIPsToKeep = 200000
+)
+
+func (vpcService *vpcService) pruneLastUsedIPAddresses(ctx context.Context, null keyedItem, tx *sql.Tx) (retErr error) {
+
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+	ctx, span := trace.StartSpan(ctx, "pruneLastUsedIPAddresses")
+	defer span.End()
+
+	logger.G(ctx).Info("Beginning purge of last used ip addresses")
+
+	result, err := tx.ExecContext(ctx,
+		`WITH addresses_by_family AS
+		    (SELECT id,
+				last_seen,
+				RANK() OVER (PARTITION BY family(ip_address), vpc_id
+							 ORDER BY last_seen DESC) AS rank
+		    FROM ip_last_used_v3
+		    WHERE last_seen != TIMESTAMP 'EPOCH')
+	    DELETE
+	    FROM ip_last_used_v3
+	    WHERE id IN
+		    (SELECT id
+		    FROM addresses_by_family
+		    WHERE rank > $1 AND last_seen > now() - INTERVAL '1 DAY')`,
```

Comment:
shouldn't this be:
`last_seen < now() - INTERVAL '1 DAY'`?

Reply:
Whoops! I updated your jira comment but didn't update the code... :(